source/opt/eliminate_dead_input_components_pass.h

```diff
@@ -57,6 +58,9 @@ class EliminateDeadInputComponentsPass : public Pass {
 
   // Change the length of the struct |struct_var| to |length|
   void ChangeStructLength(Instruction& struct_var, unsigned length);
+
+  // Process output variables instead
+  bool output_instead_;
```

Comment:
Any reason to not just rename this pass `EliminateDeadInterfaceComponentsPass`? It is definitely not intuitive that the `output` part is in the `input` file/pass as well (without the extra code mapping).

If we think the churn is too much, might be worth adding a note explaining that `Input` was here first and was easier to add this boolean then do a rename

Reply:
I will rename the pass and the files.

Reply:
Actually, I **promise** to rename the passes and files :) I have two other PRs that are already created, and renaming at this point would be a pain. The Create*() functions are the true interface for users and those are correctly named, so I don't think there is harm in delaying the renaming of the passes and files slightly.